sdks/python/apache_beam/ml/inference/run_inference.py

```diff
@@ -0,0 +1,43 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import apache_beam as beam
+from apache_beam.ml.inference.model import (
+    PytorchModel, RunInferenceModel, SklearnModel)
+
+
+# TODO
+#@beam.typehints.with_input_types(Union[_INPUT_TYPE, Tuple[_K, _INPUT_TYPE]])
+#@beam.typehints.with_output_types(Union[_OUTPUT_TYPE, Tuple[_K, _OUTPUT_TYPE]])
+class RunInference(beam.PTransform):
+  def __init__(self, model: RunInferenceModel, batch_size=None, **kwargs):
+    self._model = model
+    self._batch_size = batch_size
+
+  def expand(self, examples: beam.PCollection) -> beam.PCollection:
+
+    if isinstance(self._model, PytorchModel):
```

Comment:
Generally isinstance chains like this is an anti-pattern. RunInferenceModel should have a sufficiently expressive interface that the implementation-specific can be fully encapsulated in the various subclass implementations.

Reply:
👍